plugins/node/opentelemetry-instrumentation-memcached/package.json

```diff
@@ -0,0 +1,71 @@
+{
+  "name": "@opentelemetry/instrumentation-memcached",
+  "version": "0.22.0",
+  "description": "OpenTelemetry memcached automatic instrumentation package.",
+  "main": "build/src/index.js",
+  "types": "build/src/index.d.ts",
+  "repository": "open-telemetry/opentelemetry-js-contrib",
+  "scripts": {
+    "clean": "rimraf build/*",
+    "codecov": "nyc report --reporter=json && codecov -f coverage/*.json -p ../../",
+    "compile": "npm run version:update && tsc -p .",
+    "lint": "eslint . --ext .ts",
+    "lint:fix": "eslint . --ext .ts --fix",
+    "precompile": "tsc --version",
+    "prepare": "npm run compile",
+    "tdd": "npm run test -- --watch-extensions ts --watch",
+    "test": "nyc ts-mocha -p tsconfig.json 'test/**/*.test.ts'",
+    "test:debug": "cross-env RUN_MEMCACHED_TESTS_LOCAL=true ts-mocha --inspect-brk --no-timeouts -p tsconfig.json 'test/**/*.test.ts'",
+    "test:local": "cross-env RUN_MEMCACHED_TESTS_LOCAL=true npm run test",
+    "version:update": "node ../../../scripts/version-update.js"
+  },
+  "keywords": [
+    "opentelemetry",
+    "memcached",
+    "nodejs",
+    "tracing",
+    "profiling",
+    "instrumentation"
+  ],
+  "author": "OpenTelemetry Authors",
+  "license": "Apache-2.0",
+  "engines": {
+    "node": ">=8.0.0"
+  },
+  "files": [
+    "build/src/**/*.js",
+    "build/src/**/*.d.ts",
+    "doc",
+    "LICENSE",
+    "README.md"
+  ],
+  "publishConfig": {
+    "access": "public"
+  },
+  "peerDependencies": {
+    "@opentelemetry/api": "^1.0.1"
+  },
+  "devDependencies": {
+    "@opentelemetry/api": "1.0.1",
+    "@opentelemetry/context-async-hooks": "0.22.0",
+    "@opentelemetry/node": "0.22.0",
+    "@opentelemetry/test-utils": "^0.22.0",
+    "@opentelemetry/tracing": "0.22.0",
+    "@types/mocha": "7.0.2",
+    "@types/node": "14.17.2",
+    "codecov": "3.8.2",
+    "cross-env": "7.0.3",
+    "gts": "3.1.0",
+    "memcached": "^2.2.2",
+    "mocha": "7.2.0",
+    "nyc": "15.1.0",
+    "rimraf": "3.0.2",
+    "ts-mocha": "8.0.0",
+    "typescript": "4.3.2"
+  },
+  "dependencies": {
+    "@opentelemetry/instrumentation": "^0.22.0",
+    "@opentelemetry/semantic-conventions": "^0.22.0",
+    "@types/memcached": "^2.2.6"
```

Comment:
Yes. This is the current practice. See references here: https://github.com/open-telemetry/opentelemetry-js/pull/2256

Reply:
since typescript 4+ if you use this instrumentation without the main package that was instrumented and types it will fail, strange behaviour but this is what we have, before ts 4 it was working fine as types could be references from dev

Reply:
When you define a type inside the `BaseInstrumentation` generic (to "try" to have type completion when patching/unpatching, it creates some sort of link that typescript will by default try to verify. If you don't bring the types as dependency, typescript will complain unless you add `skipLibCheck: true`

Reply:
Thanks @vmarchaud @obecny 
Wasn't aware of it.
I Will fix it in my repo as well